tests/rptest/tests/group_membership_test.py

```diff
@@ -235,12 +235,19 @@ def check_metric():
             assert metric_key in metrics_offsets
             assert metrics_offsets[metric_key] == 50
 
-            self.client().delete_topic(topic)
-
         check_metric()
-        metrics_offsets = self._get_offset_from_metrics(group)
-        assert metrics_offsets is None
+
+        self.client().delete_topic(topic)
+        wait_until(lambda: self._get_offset_from_metrics(group) is None,
+                   timeout_sec=10,
+                   backoff_sec=1,
+                   err_msg="Topic metrics haven't dissapeared")
+
         self.client().create_topic(topic_spec)
+        wait_until(lambda: topic_spec.name in rpk.list_topics(),
+                   timeout_sec=10,
+                   backoff_sec=1,
+                   err_msg="Topic is not recreated")
```

Comment:
@ZeDRoman why are you using `list_topics` here? can we not use wait_until on the exact condition we care about like we did above?